### ISSUE TITLE
Support for 'interpolate'

### ISSUE THREAD

**Issue by ctjacobs** [firedrakeproject/firedrake#485]
In DOLFIN I can do:

```
from dolfin import *
mesh = UnitSquareMesh(2, 2)
fs = FunctionSpace(mesh, "CG", 1)
fs2 = FunctionSpace(mesh, "CG", 2)

f = interpolate(Expression("2.0"), fs)
g = interpolate(Constant(3.0), fs)
h = interpolate(g, fs2)

print f.vector().array()
print g.vector().array()
print h.vector().array()
```

which gives

```
[ 2.  2.  2.  2.  2.  2.  2.  2.  2.]
[ 3.  3.  3.  3.  3.  3.  3.  3.  3.]
[ 3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.  3.
  3.  3.  3.  3.  3.  3.  3.]
```

While Firedrake does of course have the `interpolate` method as part of the `Function` class which can interpolate `Expression`s, it has no stand-alone `interpolate` function which can take an `Expression`, `Constant` or `Function` and return another `Function`:

```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    f = interpolate(Expression("2.0"), fs)
NameError: name 'interpolate' is not defined
```

This is needed for Firedrake-adjoint stuff and to be consistent with the DOLFIN interface.

**Comment by ctjacobs** [comment]
Partially addressed in #489 for the case of `Expression`s.

**Comment by colinjcotter** [comment]
This would also be useful for writing higher order fields to vtu, since interpolate is more suitable than project, usually.

**Comment by miklos1** [comment]
Fixed by #715.
